connectors/db/vector.py

```diff
@@ -30,6 +35,52 @@
     "",
 ]
 
+# because we currently cannot access usage_metadata for embedding calls nor use
+# get_openai_callback() in the same way we can for chat model calls...
+# (see https://github.com/langchain-ai/langchain/issues/945)
+#
+# we use a work-around inspired by https://github.com/encode/httpx/discussions/3073
+
+
+class CustomResponse(httpx.Response):
```

Comment:
I looked at this for a long time because subclassing a big, core class like Response can be scary. I looked at httpx.Response and its a big class with a lot of complexity and a lot of internal state. So I read a bunch about HTTPX, got my mind around the base class, and went over this code line by line and how it would interact with the base class method. This looks totally safe, and quite clever.